### PR TITLE
bump Python from 3.8 to 3.9

### PR DESCRIPTION
**Pull request by goneri**

### CODE REVIEW

**Comment by felixfontein** [comment]
This PR does not fix the underlying problem (networking between the containers does not properly work with podman), but instead modifies the images so that no networking is needed with the latest ansible-core version.

We still have to decide how to continue with the CentOS Stream 8 image. We can either a) switch from Py 3.8 to 3.9, b) include both Py 3.8 and 3.9, or c) have two separate images, one with Py 3.8 and one with Py 3.9. I'm not sure what would be the best solution...

CC @dmsimard

**Comment by dmsimard** [comment]
> We still have to decide how to continue with the CentOS Stream 8 image. We can either a) switch from Py 3.8 to 3.9, b) include both Py 3.8 and 3.9, or c) have two separate images, one with Py 3.8 and one with Py 3.9. I'm not sure what would be the best solution...

Testing a matrix of every possible distro, python version, ansible-core and ansible is challenging for a number of reasons :p

For context, we initially came up with the centos8-stream image when ansible-core stopped testing centos8 (due to EOL) and then decided against testing centos8-stream.
I felt it was important to keep this testing coverage, even in some limited capacity: there are Ansible users on centos8-stream and it also happens to be ahead of RHEL 8, giving us a heads up in case there is an issue.

Anyway, I think we have to be reasonable and test what we can but that doesn't necessarily mean testing everything.
We could call that "best effort" which doesn't result in a lot of maintenance work while giving us a good return on investment in terms of time and effort to provide a minimum amount of coverage.

Considering python3.8 is likely to be tested in a number of other scenarios and pipelines, we could probably get away with bumping to python3.9.

Does that make sense ?

File: execution-environments/2.12-with-ansible5/bindep.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
-python38 [platform:rpm]
-python38-devel [platform:rpm]
-python38-pip [platform:rpm]
+python39 [platform:rpm]
+python39-devel [platform:rpm]
+python39-pip [platform:rpm]
```
Comment (dmsimard) [inline execution-environments/2.12-with-ansible5/bindep.txt:3]: This EE is outdated by now -- it was meant for ansible-core 2.12 and ansible 5 and was last published 9 months ago: https://quay.io/repository/ansible-community/test-ee?tab=tags

I am not aware if it is actually used anywhere but we're up to ansible-core 2.14 ans ansible 7 now.

We should probably leave this one alone and create a 2.14/7 one if necessary ?

**Comment by felixfontein** [comment]
Since the CentOS Stream 8 image has already been updated in other PRs (Python 3.8 deprecated, Python 3.6 (system) and 3.9 added), and the EE is pretty much dead and unused, I'm closing this. Thanks for creating this PR @goneri! And thanks for your comments @dmsimard!

**Pull request closed without merging**